src/hotspot/os/windows/os_windows.cpp

```diff
@@ -2683,91 +2608,11 @@ LONG WINAPI topLevelExceptionFilter(struct _EXCEPTION_POINTERS* exceptionInfo) {
-      // page is not, the pc and the faulting address might be slightly
-      // different - we still want to unguard the 2nd page in this case.
-      //
-      // 15 bytes seems to be a (very) safe value for max instruction size.
-      bool pc_is_near_addr =
-        (pointer_delta((void*) addr, (void*) pc, sizeof(char)) < 15);
-      bool instr_spans_page_boundary =
-        (align_down((intptr_t) pc ^ (intptr_t) addr,
-                         (intptr_t) page_size) > 0);
-
-      if (pc == addr || (pc_is_near_addr && instr_spans_page_boundary)) {
-        static volatile address last_addr =
-          (address) os::non_memory_address_word();
-
-        // In conservative mode, don't unguard unless the address is in the VM
-        if (UnguardOnExecutionViolation > 0 && addr != last_addr &&
-            (UnguardOnExecutionViolation > 1 || os::address_is_in_vm(addr))) {
-
-          // Set memory to RWX and retry
-          address page_start = align_down(addr, page_size);
-          bool res = os::protect_memory((char*) page_start, page_size,
-                                        os::MEM_PROT_RWX);
-
-          log_debug(os)("Execution protection violation "
-                        "at " INTPTR_FORMAT
-                        ", unguarding " INTPTR_FORMAT ": %s", p2i(addr),
-                        p2i(page_start), (res ? "success" : os::strerror(errno)));
-
-          // Set last_addr so if we fault again at the same address, we don't
-          // end up in an endless loop.
-          //
-          // There are two potential complications here.  Two threads trapping
-          // at the same address at the same time could cause one of the
-          // threads to think it already unguarded, and abort the VM.  Likely
-          // very rare.
-          //
-          // The other race involves two threads alternately trapping at
-          // different addresses and failing to unguard the page, resulting in
-          // an endless loop.  This condition is probably even more unlikely
-          // than the first.
-          //
-          // Although both cases could be avoided by using locks or thread
-          // local last_addr, these solutions are unnecessary complication:
-          // this handler is a best-effort safety net, not a complete solution.
-          // It is disabled by default and should only be used as a workaround
-          // in case we missed any no-execute-unsafe VM code.
-
-          last_addr = addr;
-
-          return EXCEPTION_CONTINUE_EXECUTION;
-        }
-      }
-
-      // Last unguard failed or not unguarding
-      tty->print_raw_cr("Execution protection violation");
-#if !defined(USE_VECTORED_EXCEPTION_HANDLING)
-      report_error(t, exception_code, addr, exception_record,
-                   exceptionInfo->ContextRecord);
-#endif
-      return EXCEPTION_CONTINUE_SEARCH;
-    }
-  }
-#endif // _WIN64
-
-#if defined(_M_AMD64) || defined(_M_IX86)
+#if defined(_M_AMD64)
```

Comment:
The check for LP64 on line 2622 below seems redundant now

Reply:
Indeed, nice catch! I also found another place in this file that were guarded by `_LP64` that I removed. I also did a grep on `LP64` in `hotspot/os/windows`, but there were no more instances.